src/grpc/stream_infer_handler.cc

```diff
@@ -573,17 +573,19 @@ ModelStreamInferHandler::StreamInferResponseComplete(
   }
 #endif  // TRITON_ENABLE_TRACING
 
-  // Log appropriate errors
   bool is_complete =
       state->complete_ || (flags & TRITONSERVER_RESPONSE_COMPLETE_FINAL) != 0;
+
+  // Log appropriate messages
   if (!state->is_decoupled_) {
     if (!is_complete) {
-      LOG_ERROR << "[INTERNAL] ModelStreamInfer received a response without "
-                   "FINAL flag for a model with one-to-one transaction";
+      LOG_VERBOSE(2)
```

Comment:
Why did we change `LOG_ERROR` to `LOG_VERBOSE(2)` ?

Reply:
It's no longer an error to receive response and final_flag as two separate responses for non-decoupled models, but assuming we're keeping the log for some visibility or test log grepping.

Reply:
Currently we "require" the final flag to be sent with the final/only response for non-decoupled streaming, but this is not an error because the model may send the response first and then the final flag. I change it to `LOG_VERBOSE(2)` because it is not an error, but kept the log for debugging purpose.

Reply:
How would frontend handle cases where multiple responses are sent in a non-decoupled case? 
I would suggest changing the message to something more generic if we are keeping it for debugging purpose. For example enhance the verbose message printed out in L564.

Reply:
> How would frontend handle cases where multiple responses are sent in a non-decoupled case?

I think the extra responses go into the `response_queue_`, but only the [first response will be written](https://github.com/triton-inference-server/server/blob/main/src/grpc/infer_handler.h#L921). Then, the extra responses are discarded when the [state resets](https://github.com/triton-inference-server/server/blob/main/src/grpc/infer_handler.h#L1048).

Reply:
> I would suggest changing the message to something more generic if we are keeping it for debugging purpose. For example enhance the verbose message printed out in L564.

Good idea! Updated: [Enhance logging for non-decoupled on null response and final flag](https://github.com/triton-inference-server/server/pull/7265/commits/42a6a80af9ad6eb9b7c54fb3f7338b060b2f6010)

Reply:
Can we explicitly return an error rather ignoring the second response? It can also be part of the additional core checks that we were targeting.

Reply:
This is a good idea! I don't think it is hard to implement because we should be able to remove the first response from the queue and replace it with an error response. Then, wait for the complete final flag and sent the response.

@Tabrizian @rmccorm4 Can we do it when we are working on additional core checks?
It is because ignoring the second response is the current behavior not introduced by this change. I want to keep the scope of this PR small and only acknowledge the current behavior on the test case, plus the original intent on fixing the issue where the final flag can be sent as a flag only response after the first response.

Reply:
If this is existing behavior then I'm good with a separate follow-up. Thanks for the explanation!